### PR TITLE
[SYCL][COMPAT] Add funnelshift APIs and corresponding tests

### PR DESCRIPTION
**Pull request by joeatodd**
Adds `funnelshift_*` math functions to SYCLcompat, and corresponding tests.

### CODE REVIEW

File: sycl/doc/syclcompat/README.md [hunk 1]
```diff
@@ -1457,6 +1457,15 @@ static void invoke_kernel_function(kernel_function &function,
 
 ### Math Functions
 
+The `funnelshift_*` APIs take two unsigned integer arguments (`low` and `high`),
+concatenate them, bit-shift the 64-bit result by `shift` bits and return either
+the most or least significant 32-bits. The `_l*` variants shift left and return
```
Comment (AidanBeltonS) [inline sycl/doc/syclcompat/README.md:1462]: First sentence very wordy and hard to follow. There is likely a much easier way to explain this operation. It should probably also be 2 sentences

**Review by AidanBeltonS — APPROVED** [review]
One comment, otherwise looks fine

File: sycl/doc/syclcompat/README.md (hunk 1 above)
Comment (joeatodd) [reply sycl/doc/syclcompat/README.md]: Good feedback, thanks @AidanBeltonS. I've addressed this now.

**Comment by joeatodd** [comment]
@intel/llvm-gatekeepers this is ready to merge :pray:
